webapp/channels/src/components/advanced_text_editor/use_key_handler.tsx

```diff
@@ -0,0 +1,368 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type React from 'react';
+import {useCallback, useEffect, useRef} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {getBool} from 'mattermost-redux/selectors/entities/preferences';
+
+import {emitShortcutReactToLastPostFrom} from 'actions/post_actions';
+import {editLatestPost} from 'actions/views/create_comment';
+
+import type {TextboxElement} from 'components/textbox';
+import type TextboxClass from 'components/textbox/textbox';
+
+import Constants, {Locations, Preferences} from 'utils/constants';
+import * as Keyboard from 'utils/keyboard';
+import {type ApplyMarkdownOptions} from 'utils/markdown/apply_markdown';
+import {pasteHandler} from 'utils/paste';
+import {isWithinCodeBlock, postMessageOnKeyPress} from 'utils/post_utils';
+import * as UserAgent from 'utils/user_agent';
+import * as Utils from 'utils/utils';
+
+import type {GlobalState} from 'types/store';
+import type {PostDraft} from 'types/store/draft';
+
+const KeyCodes = Constants.KeyCodes;
+
+const useKeyHandler = (
+    draft: PostDraft,
+    channelId: string,
+    postId: string,
+    caretPosition: number,
+    isValidPersistentNotifications: boolean,
+    location: string,
+    textboxRef: React.RefObject<TextboxClass>,
```

Comment:
Instead of a ref, this seems like we could get the textbox from `e.target` to avoid passing that around

Reply:
As mentioned below, we're not changing this for now to minimize the additional changes here